CodeGeneration/Sources/generate-swiftsyntax/SyntaxBaseNodesFile.swift

```diff
@@ -0,0 +1,228 @@
+                    SwitchCase(
+                      label: .case(SwitchCaseLabel {
+                        for childNode in SYNTAX_NODES where childNode.baseKind == node.syntaxKind {
+                          CaseItem(
+                            pattern: EnumCasePatternSyntax(
+                              period: .period,
+                              caseName: .identifier(childNode.swiftSyntaxKind))
+                          )
+                        }
+                      })) {
+                        BreakStmt()
+                      }
+                    
+                    SwitchCase("default:") {
+                      FunctionCallExpr("fatalError(\"Unable to create \(raw: node.name) from \\(data.raw.kind)\")")
+                    }
+                  }
+              })
+            )
+          }
+        )
+        
+        Expr("self._syntaxNode = Syntax(data)")
+      }
+      
+      FunctionDecl("""
+        public func `is`<S: \(raw: node.name)Protocol>(_ syntaxType: S.Type) -> Bool {
+          return self.as(syntaxType) != nil
+        }
+        """)
+      
+      FunctionDecl("""
+        public func `as`<S: \(raw: node.name)Protocol>(_ syntaxType: S.Type) -> S? {
+          return S.init(self)
+        }
+        """)
+      
+      FunctionDecl("""
+        public func cast<S: \(raw: node.name)Protocol>(_ syntaxType: S.Type) -> S {
+          return self.as(S.self)!
+        }
+        """)
+      
+      FunctionDecl("""
+        /// Syntax nodes always conform to `\(raw: node.name)Protocol`. This API is just
+        /// added for consistency.
+        /// Note that this will incur an existential conversion.
+        @available(*, deprecated, message: "Expression always evaluates to true")
+        public func isProtocol(_: \(raw: node.name)Protocol.Protocol) -> Bool {
+          return true
+        }
+        """)
+      
+      FunctionDecl("""
+        /// Return the non-type erased version of this syntax node.
+        /// Note that this will incur an existential conversion.
+        public func asProtocol(_: \(raw: node.name)Protocol.Protocol) -> \(raw: node.name)Protocol {
+          return Syntax(self).asProtocol(\(raw: node.name)Protocol.self)!
+        }
+        """)
+      
+      
+      VariableDecl(
+        modifiers: [DeclModifier(name: .public), DeclModifier(name: .static)],
+        name: IdentifierPattern("structure"),
+        type: TypeAnnotation(
+          type: TypeSyntax("SyntaxNodeStructure"))
+      ) {
+        ReturnStmt(
+          expression: FunctionCallExpr(
+            callee: MemberAccessExpr(".choices")) {
+              TupleExprElement(
+                expression: ArrayExprSyntax {
+                  for childNode in SYNTAX_NODES where childNode.baseKind == node.syntaxKind {
+                    ArrayElement(
+                      expression: FunctionCallExpr("\n.node(\(raw: childNode.name).self)"),
+                      trailingComma: .comma)
```

Comment:
IIRC this trailing comma should also be inferred by SwiftSyntaxBuilder

Reply:
Once again your are right! 🚀